HW-7 ViewModel 적용

BeokBeok opened the pull request:
View Model 과 Sort 기능 적용 하였습니다.

BeokBeok committed:
HW-7 중복된 dependency 제거

BeokBeok committed:
HW-7 life cycle extension 관련 dependency 추가

BeokBeok committed:
HW-7 AAC ViewModel 적용

BeokBeok committed:
HW-7 it 을 중복으로 사용하지 않도록 변경

BeokBeok committed:
HW-7 trading value 관련 변수명 변경

BeokBeok committed:
HW-7 쓰레드 환경에 맞는 live data set 함수 변경

BeokBeok committed:
HW-7 Model 관련 패키지 정리

BeokBeok committed:
HW-7 Ticker 관련 변수명 통일

BeokBeok committed:
HW-7 gradle 업데이트

BeokBeok committed:
HW-7 Ticker 관련 변수명 통일로 인한 함수명 변경

BeokBeok committed:
HW-7 Room 관련 dependency 추가

BeokBeok committed:
HW-7 LocalDataSource 적용

BeokBeok committed:
HW-7 Market 외에 필터링이 되지 않는 현상 수정

BeokBeok committed:
HW-7 sort 함수 구현

- 클릭 시 데이터는 정렬되지만, adapter 에 갱신되지 않음

BeokBeok committed:
HW-7 CompositeDisposible 을 dispose 하는 위치를 Base 에서 하도록 변경

BeokBeok committed:
HW-7 Single Line 구현부 코드 정리

BeokBeok committed:
HW-7 RxEventBus 구현 및 이중 호출을 방지하기 위한 prefix 데이터 추가

BeokBeok committed:
HW-7 미사용 Resource 제거

BeokBeok committed:
HW-7 prefix 관련 Column 제거

BeokBeok committed:
HW-7 CompositeDisposable 을 dispose 하는 시점 변경

onDestroy -> onDestroyView

BeokBeok committed:
HW-7 View Pager Limit 설정

BeokBeok committed:
HW-7 정렬이 모든 마켓 타입에 대하여 일괄적으로 적용되도록 변경

BeokBeok committed:
HW-7 오름차순, 내림차순 정렬 기능 적용

BeokBeok committed:
HW-7 정렬 아이콘 적용

NackUn reviewed (dismissed):
고생하셨습니다!
93a8ca6 ROOM을 사용해보지 않아서 리뷰가 부족하네요 ㅠ

sjjeong reviewed (changes requested):
CompositeDisposable를 왜 매번 생성해서 사용하셨나요?
객체가 달라져서 사용하는 의미가 없는것 같습니다.

BeokBeok committed:
HW-7 CompositeDisposable Object 추가

BeokBeok commented:
@sjjeong 
CompositeDisposable 객체를 만들고 작업한 것으로 착각했네요.
말씀 안해주셨으면 계속 모르고 넘어갔...

감사합니다! 추가 하였습니다.

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/util/CompositeDisposable.kt:
```diff
@@ -0,0 +1,20 @@
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.disposables.Disposable
+
+object CompositeDisposable {
```
예전에 현성님꺼 리뷰에서인가 이렇게 하셔서 제가 문제가 발생할 수 있다고 말한적이 있었어요.
CompositeDisposable은 각각 클래스에서 따로 객체로 관리해야해요

BeokBeok replied:
94a14be9fac410cb9984229b77084b1a7f892fb6
뷰모델에서 일괄적으로 관리하도록 변경하였습니다.

Commit:
HW-7 CompositeDisposable 을 각 클래스에서 생성하도록 변경

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/136#discussion_r307797299

Commit:
HW-7 CompositeDisposable 을 ViewModel 에서 관리하도록 변경

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/136#discussion_r307797299
